### PR TITLE
Support for TPS

### PR DESCRIPTION
**Pull request by gsr-shanks**
Initial revision for support for TPS. This is still WIP.

### CODE REVIEW

File: fedmsg_meta_umb/tests/test_tps.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# Authors:  Gowrishankar Rajaiyan <grajaiya@redhat.com>
+
+import fedmsg.tests.test_meta
+
+
+class TestTPSStart(fedmsg.tests.test_meta.Base):
+    """ TPS performs basic package sanity testing of rpms.
+
+    Messages (like the example given here) are published when an **TPS**
+    job is **starting**. (A TPS job performs basic sanity testing for the given brew build).
```
Comment (ralphbean) [inline fedmsg_meta_umb/tests/test_tps.py]: Can you use reStructuredText format to provide a link to TPS documentation here? (high level "what is TPS?" documentation.)

File: fedmsg_meta_umb/tests/test_tps.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    """
+    expected_title = 'tps.starting'
+    expected_packages = set(['systemd'])
```
Comment (ralphbean) [inline fedmsg_meta_umb/tests/test_tps.py]: This is a good start!

I'd recommend getting `expected_subti` and `expected_link` working before we merge this.

`expected_icon` would be nice to have, but not strictly required.

**Comment by gsr-shanks** [comment]
@ralphbean Thanks for the review. Any changes that I need to do for travis-ci to pass ? The assert seems to fail.

**Comment by ralphbean** [comment]
> Any changes that I need to do for travis-ci to pass ? The assert seems to fail.

Yeah.  Although, from a read it is not clear to me why they're not passing.

**Comment by ralphbean** [comment]
Ah, it took me a minute to find it, but here it is:

```python
ERROR:fedmsg:Must declare a __description__
Traceback (most recent call last):
  File "/home/threebean/.virtualenvs/fedmsg_meta_umb/lib/python2.7/site-packages/fedmsg-0.18.3-py2.7.egg/fedmsg/meta/__init__.py", line 109, in make_processors
    processors.append(processor.load()(_, **config))
  File "/home/threebean/.virtualenvs/fedmsg_meta_umb/lib/python2.7/site-packages/fedmsg-0.18.3-py2.7.egg/fedmsg/meta/base.py", line 77, in __init__
    raise ValueError("Must declare a __description__")
ValueError: Must declare a __description__
```

**Comment by ralphbean** [comment]
Add the same kind of `__attribute__` attributes that the other processors have.

File: fedmsg_meta_umb/tests/test_tps.py (hunk 1 above)
Comment (gsr-shanks) [reply fedmsg_meta_umb/tests/test_tps.py]: Link provided in rST format.

File: fedmsg_meta_umb/tests/test_tps.py (hunk 2 above)
Comment (gsr-shanks) [reply fedmsg_meta_umb/tests/test_tps.py]: missing attributes and messages added.

**Comment by gsr-shanks** [comment]
@ralphbean, looks all set to me. Please review.

File: fedmsg_meta_umb/tests/test_tps.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    """ `TPS-next <https://rpm-factory-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/test-package-sanity-documentation/
+    lastSuccessfulBuild/artifact/docs/index.html>`_ performs basic package sanity testing of rpms.
+
+    Messages (like the example given here) are published when an **TPS**
+    job is **performed**. (A TPS job performs basic sanity testing for the given brew build).
+    """
+    expected_title = 'tps'
+    expected_link = ('https://rpm-factory-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/'
+                     'test-package-sanity-development/label=ose-slave-tps,provision_arch=x86_64/1915/')
+    expected_subti = 'package sanity testing results for component: setup-2.8.71-7.el7_4 with brew task_id: 14393825'
```
Comment (ralphbean) [inline fedmsg_meta_umb/tests/test_tps.py]: A soft recommendation:  put the value of `tps_status` somewhere in the subtitle for this one.

Something like:  `'package sanity testing PASSED for component: setup....'`.  What do you think?  Human readers will be able to see what they want to know without having to click through.

**Pull request merged**